### PR TITLE
 add hive in CI

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

Resolves: #805 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: docker/hive/Dockerfile [hunk 1]
```diff
@@ -39,7 +39,93 @@ RUN git clone -v "https://github.com/kkrt-labs/kakarot-indexer.git"
 FROM mongo:6.0.8 as mongo
 
 # Ethereum RPC Server
-FROM ghcr.io/kkrt-labs/kakarot-rpc/node:v0.2.4 as rpc
+### Build the Cairo programs artifacts
```
Comment (Eikix) [inline docker/hive/Dockerfile:42]: Weirdly this does not work anymore!

```
FROM ghcr.io/kkrt-labs/kakarot-rpc/node:v0.2.4 as rpc
``` 
Will allow hive tests to work locally,

But replacing it by the actual build will make it fail instantly

Comment (Eikix) [reply docker/hive/Dockerfile:42]: For now I literally copied part of the RPC docker file to see if it worked

Comment (Eikix) [reply docker/hive/Dockerfile:42]: I FIXED IT

File: docker/rpc/Dockerfile [hunk 2]
```diff
@@ -22,10 +22,7 @@ RUN curl -L https://foundry.paradigm.xyz -o foundry.sh \
     unzip \
     zip
 
-COPY .git ./.git
-COPY .gitmodules .gitmodules
-COPY Makefile Makefile
-COPY scripts scripts
+COPY . .
```
Comment (greged93) [inline docker/rpc/Dockerfile:25]: I think we went to avoid copying the whole context into the image.

**Review by greged93 — APPROVED** [review]

**Pull request merged**